packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -112,7 +113,8 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
               aria-haspopup='true'
               autoCapitalize='off'
               autoComplete='off'
-              role='combobox' />
+              role='combobox'
+              disabled={ this.state.items.length >= itemLimit } />
```

Comment:
Not sure how hard the implementation would be but not allowing items > itemLimit but still keeping the control editable would work. Similar to HTML native `<input maxLength= '10'>` .